tools/vz/cmd/analyze/analyze_test.go

```
@@ -26,12 +25,9 @@ const istioIPErr = "Verrazzano install failed as no IP found for service istio-i
 // WHEN I call cmd.Execute from read only dir with a valid capture-dir and report-format set to "summary"
 // THEN expect the command to do the analysis and generate report file into tmp dir
 func TestAnalyzeDefaultFromReadOnlyDir(t *testing.T) {
-	stdoutFile, stderrFile := createStdTempFiles(t)
-	defer func() {
-		os.Remove(stdoutFile.Name())
-		os.Remove(stderrFile.Name())
-	}()
-	rc := helpers.NewFakeRootCmdContext(genericclioptions.IOStreams{In: os.Stdin, Out: stdoutFile, ErrOut: stderrFile})
+	rc, err := helpers.NewFakeRootCmdContextWithFiles()
+	assert.Nil(t, err)
```

should the defer be done before the assert here

---

Or if we can pass "t" into the helper there and have it assert when it fails rather than returning an error at all. Then we can get rid of all of the "assert.Nil(t, err)"'s

---

I think that passing "t" into the helper would work to further is a great idea! I will work on implementing that!